File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int solution(string word) {
    int answer = 0;
    
    vector<char> vec = {' ','A', 'E', 'I', 'O', 'U'};
    
    int len = word.length();
    for(int i=0; i< len;i++) {
        int idx = find(vec.begin(), vec.end(), word[i]) - vec.begin();
        
        answer += idx;
        
        int sum = 0;
        for(int j=4-i; j>0; j--) {
            sum += pow(5,j);
        }
        
        answer += sum*(idx-1);
        
    }
    
    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    var answer = 0;
    
    const letters = ['', 'A', 'E', 'I','O','U'];
    
    const len = word.length;
    for(let i=0; i< len;i++) {
        const idx = letters.indexOf(word[i]);
        
        answer += idx;
        
        let sum = 0;
        for(let j=4-i; j>0; j--) {
            sum += Math.pow(5,j);
        }
        
        answer += sum * (idx-1);
    }
    
    return answer;
}